core/transfer/src/main/java/org/eclipse/dataspaceconnector/transfer/core/transfer/TransferProcessManagerImpl.java

```diff
@@ -570,42 +574,77 @@ private void processProviderRequest(TransferProcess process, DataRequest dataReq
         }
     }
 
+    private boolean processConsumerRequest(TransferProcess process, DataRequest dataRequest) {
+        int retryCount = process.getStateCount() - 1;
+        if (retryCount > 0) {
```

Comment:
This logic seems too complicated to handle it here, because it make also testing more difficult.
I would advice to introduce an interface like (names are just the first proposal, could be better for sure :) )
```java
private interface SendEntityRetryManager<T> {
    boolean shouldWaitNextTick(T entity);
}
```
(generic because this will likely be implemented for `ContractNegotiation` and probably others).

doing so the `processConsumerRequest` body will become only
```java
if (sendEntityRetry.shouldWaitNextTick(process)) {
    // Break lease
    transferProcessStore.update(process);

    return false;
} else {
    sendConsumerRequest(process, dataRequest);
    return true;
}
```
that's more easily testable, and then all the logic and the clock will be moved in a `TransferProcess` implementation of the interface, and one instance of that will become a collaborator of `TransferProcessManager` and it would become usable also in other parts of the manager if needed.

Reply:
Implemented